staging/src/k8s.io/client-go/tools/remotecommand/remotecommand_test.go

```diff
@@ -190,6 +212,29 @@ func attach2Server(rawURL string, options StreamOptions) error {
 	}
 }
 
+func attach2ServerForAWhile(rawURL string, options StreamOptions) error {
+	uri, _ := url.Parse(rawURL)
+	exec, terminate, err := NewTerminableSPDYExecutor(&rest.Config{Host: uri.Host}, "POST", uri)
+	if err != nil {
+		return err
+	}
+
+	e := make(chan error)
+	go func(e chan error) {
+		e <- exec.Stream(options)
+	}(e)
+	go func() {
+		time.Sleep(time.Second)
+		terminate()
```

Comment:
How do we know whether the `terminate()` successfully closed the connection before `exec.Stream(options)` returns? From the code I cannot tell how long `exec.Stream(options)` takes.

Reply:
The exec.Stream(options) will never return because the stdin/stdout/stderr never exits (which is exactly what this patch deal with).
In this test case,  if the terminate() does not work as we expect,  `case <-time.After(wait.ForeverTestTimeout)` will be triggered (ForeverTestTimeout is 30s) and this test case does not pass.